File: app/src/main/java/me/echeung/moemoekyun/api/endpoints/Jpop.java
package me.echeung.moemoekyun.api.endpoints;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Jpop extends Library {

    public static final Library INSTANCE = new Jpop();

    public static final String NAME = "jpop";

    public String getName() {
        return NAME;
    }

    public String getSocketUrl() {
        return "wss://listen.moe/gateway";
    }

    public String getStreamUrl() {
        return "https://listen.moe/fallback";
    }

}

File: app/src/main/java/me/echeung/moemoekyun/client/api/v5/library/Jpop.kt
package me.echeung.moemoekyun.client.api.v5.library

class Jpop private constructor() : Library(
        NAME,
        "wss://listen.moe/gateway",
        "https://listen.moe/fallback") {
    companion object {
        const val NAME = "jpop"

        val INSTANCE: Library = Jpop()
    }
}
